hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDeploymentManagement.java

```diff
@@ -417,12 +428,12 @@ private void setAssignedDistributionSetAndTargetUpdateStatus(final AbstractDsAss
         assignmentStrategy.setAssignedDistributionSetAndTargetStatus(set, targetIdsChunks, currentUser);
     }
 
-    private List<JpaAction> createActions(final Collection<TargetWithActionType> targetsWithActionType,
-            final List<JpaTarget> targets, final AbstractDsAssignmentStrategy assignmentStrategy,
-            final JpaDistributionSet set) {
+    private List<JpaAction> createActions(final String triggeredBy, final Collection<TargetWithActionType> targetsWithActionType,
+            final List<JpaTarget> targets, final AbstractDsAssignmentStrategy assignmentStrategy, final JpaDistributionSet set) {
 
         return targetsWithActionType.stream().map(twt -> assignmentStrategy.createTargetAction(twt, targets, set))
-                .filter(Objects::nonNull).map(actionRepository::save).collect(Collectors.toList());
+                .filter(Objects::nonNull).peek(a -> a.setTriggeredBy(triggeredBy))
```

Comment:
Why don't you set `triggeredBy` within the strategy? Modification after action has been created is not a good idea if someone else uses this method